waspc/src/Wasp/SemanticVersion.hs

```diff
@@ -0,0 +1,31 @@
+module Wasp.SemanticVersion
+  ( SemanticVersion (..),
+    VersionBounds (..),
+    isVersionInBounds,
+  )
+where
+
+import Text.Printf (printf)
+
+data SemanticVersion = SemanticVersion
+  { major :: Int,
+    minor :: Int,
+    patch :: Int
+  }
+  deriving (Eq, Ord)
+
+instance Show SemanticVersion where
+  show (SemanticVersion mjr mnr ptc) = printf "%d.%d.%d" mjr mnr ptc
+
+data VersionBounds
+  = Exact SemanticVersion
+  | CompatibleWith SemanticVersion
```

Comment:
I thought about adding comments here, but decided they're not needed. What do you think?

Reply:
I have to admit I don't know what `CompatibleWith` stands for :D.
Maybe it is not comments that we need, but better name?
What does it stand for? Fixed major version?

Reply:
Yes, it stands for "backwards compatible with" (i.e., same major versions). The name it a bit lacking. I was basing my naming on [this](https://semver.org/). What do you think about calling it `BackwardsCompatibleWith`?

Reply:
I ended up adding comments and linking to the documentation, better safe than sorry :)